daemon/containerd/service.go

```diff
@@ -154,9 +156,35 @@ func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID st
 	if ctr == nil {
 		return 0, 0, nil
 	}
+
+	snapshotter := i.client.SnapshotService(ctr.Driver)
+
+	usage, err := snapshotter.Usage(ctx, containerID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	imageManifest, err := getContainerImageManifest(ctr)
+	if err != nil {
+		// Best efforts attempt to pick an image.
+		// We don't have platform information at this point, so we can only
+		// assume that the platform matches host.
+		// Otherwise this will give a wrong base image size (different
+		// platform), but should be close enough.
+		mfst, err := i.GetImageManifest(ctx, ctr.Config.Image, imagetypes.GetImageOpts{})
+		if mfst == nil || err != nil {
+			// Log error, don't error out whole operation.
+			logrus.WithFields(logrus.Fields{
+				logrus.ErrorKey: err,
+				"container":     containerID,
+			}).Error("empty ImageManifest, can't calculate base image size")
```

Comment:
Perhaps make this a `Warn` instead of an error.

Reply:
We should probably write down some some kind of guideline for how to decide on warn vs error 😅

Reply:
Yeah, it's the "best attempt" part here; if we consider it an error, it should probably return an error, but if we "allow" it to fail, then perhaps a warning would make more sense. Not sure (admitted, it's sometimes a bit of a grey area).

Reply:
Changed to Warn